cloud/mutagenbox/mutagenbox.go

```diff
@@ -1,20 +1,75 @@
 package mutagenbox
 
 import (
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+	"go.jetpack.io/devbox/boxcli/featureflag"
 	"go.jetpack.io/devbox/cloud/mutagen"
 )
 
+const (
+	// relative to user home i.e. ~
+	dataDirPath = ".config/devbox/mutagen"
+)
+
 // TerminateSessionsForMachine is a devbox-specific API that calls the generic mutagen terminate API.
 // It relies on the mutagen-sync-session's labels to identify which sessions to terminate for
 // a particular machine (fly VM).
-func TerminateSessionsForMachine(machineID string, env map[string]string) error {
-	labels := MutagenSyncLabels(machineID)
+func TerminateSessionsForMachine(machineID string, userEnv map[string]string) error {
+	labels := DefaultSyncLabels(machineID)
+
+	env, err := DefaultEnv()
+	if err != nil {
+		return err
+	}
+	// the user-specified env-vars get precedence over the defaultEnvVars
+	for k, v := range userEnv {
+		env[k] = v
+	}
+
 	return mutagen.Terminate(env, labels)
 }
 
-// Ideally, this should be in the cloud package but it leads to a compile cycle.
-func MutagenSyncLabels(machineID string) map[string]string {
+func DefaultSyncLabels(machineID string) map[string]string {
 	return map[string]string{
 		"devbox-vm": machineID,
 	}
 }
+
+func DefaultEnv() (map[string]string, error) {
+	if featureflag.SSHShim.Disabled() {
+		return map[string]string{}, nil
+	}
+
+	shimDir, err := ShimDir()
+	if err != nil {
+		return nil, err
+	}
+
+	mutagenDir, err := createAndGetDataDir()
+	if err != nil {
+		return nil, err
+	}
+
+	return map[string]string{
+		"MUTAGEN_SSH_PATH":       shimDir,
+		"MUTAGEN_DATA_DIRECTORY": mutagenDir,
+	}, nil
+}
+
+// createAndGetDataDir prepares the data directory for devbox's mutagen instance
+func createAndGetDataDir() (string, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+
+	path := filepath.Join(home, dataDirPath)
+	// TODO savil. tighten the permissions.
+	if err := os.MkdirAll(path, 0777); err != nil {
```

Comment:
The `~/.mutagen` directory on my machine has 0700. Can we match that?

Reply:
fixed!